denizen_scripts/survival/repo-link/items/system/soul_item_system.dsc

```diff
@@ -308,86 +418,115 @@ item_system_build_item:
+    - if <[item].material.name.matches[<script[item_system_global_data].data_key[regex_type_check.weapon]>]>:
+      - define damage <script[item_system_global_data].data_key[defaults.damage.<[item].material.name>]>
+      - define lore <[Lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.weapons]>]>
+
+    - if <[item].material.name.matches[<script[item_system_global_data].data_key[regex_type_check.armor]>]>:
+      - define armor <script[item_system_global_data].data_key[defaults.armor.<[item].material.name>]>
+      - define lore <[Lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.armors]>]>
+
+    - foreach <list[buffs|debuffs]> as:modifier:
+      - foreach <[item].nbt[<[modifier]>].parse[to_list].combine.to_map> key:alt as:final_value:
+        - if <[alt]> == none:
+          - foreach next
+        - define lore <[Lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.buffs.<[alt]>]>]>
+
     - if <[item].has_nbt[flavor]>:
       - define flavor <[item].nbt[flavor].unescaped>
-      - define lore:|:<script[item_system_global_data].yaml_key[settings.lore.middle.flavor].parse[parsed]>
-    #- define lore:|:<script[item_system_global_data].yaml_key[settings.lore.bottom].parse[parsed]>
-    - if <[nbt]||null> != null:
-      - define final:|:nbt=<[nbt]>
-    - if <[nbt_attributes]||null> != null:
-      - define final:|:nbt_attributes=<[nbt_attributes].separated_by[$]>
-    - determine <[item].with[display_name=<[name]>;lore=<[lore]>;flags=HIDE_ALL;<[final].separated_by[;].replace[$].with[|]||nbt=vanilla/true>]>
+      - define lore <[Lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.flavor]>]>
+    - define lore <[lore].include[<script[item_system_global_data].parsed_key[settings.lore.bottom]>]>
+
+    - define NewItem <[item].with[display_name=<[name]>;lore=<[lore]>;flags=HIDE_ALL]>
+    - if !<[nbt].is_empty>:
+      - define NewItem <[NewItem].with[nbt=<[nbt]>]>
+    - else:
+      - define NewItem <[NewItem].with[nbt=vanilla/true]>
+
+    - if !<[nbt_attributes].is_empty>:
+      - define NewItem <[NewItem].with[nbt_attributes=<[nbt_attributes]>]>
+
+    - determine <[NewItem]>
     
 vanilla_craft_item_build:
   type: world
   events:
     on player crafts *_(sword|axe|chestplate|leggings|boots|helmet) bukkit_priority:HIGHEST:
-      - if <context.item.display||null> == null:
+      - if !<context.item.has_display>:
         - determine <proc[item_system_build_item].context[<context.item>]>
 
     
 vanilla_craft_item_build2:
   type: world
   events:
     on player crafts diamond_sword bukkit_priority:HIGHEST:
-      - if <context.item.display||null> == null:
-        - determine <proc[item_system_build_item].context[<context.item>]>
\ No newline at end of file
+      - if !<context.item.has_display>:
+        - determine <proc[item_system_build_item].context[<context.item>]>
```

Comment:
`ItemTag.has_display` is a cleaner way of checking whether an item's display intentionally returns invalid;
added NLAEOF